src/test/java/com/pngencoder/PngEncoderTest.java

```diff
@@ -177,89 +185,26 @@ public static IIOMetadata getImageMetaDataWithImageIO(byte[] filesBytes) throws
     }
 
     @Test
-    public void testAsChunkTypeNull() {
-        assertThrows(NullPointerException.class, () -> PngEncoderLogic.asChunk(null, new byte[1]));
-    }
-
-    @Test
-    public void testAsChunkTypeInvalid() {
-        assertThrows(IllegalArgumentException.class, () -> PngEncoderLogic.asChunk("chunk types must be four characters long", new byte[1]));
-    }
-
-    @Test
-    public void testAsChunkNullBytes() {
-        assertThrows(NullPointerException.class, () -> PngEncoderLogic.asChunk(VALID_CHUNK_TYPE, null));
-    }
-
-    @Test
-    public void testAsChunkAdds12Bytes() {
-        byte[] data = {5};
-        byte[] chunk = PngEncoderLogic.asChunk(VALID_CHUNK_TYPE, data);
-
-        int expected = 13;
-        int actual = chunk.length;
-        assertThat(actual, is(expected));
-    }
-
-    @Test
-    public void testAsChunkFirstByteIsSizeOfData() {
-        byte[] data = {5, 7};
-        byte[] chunk = PngEncoderLogic.asChunk(VALID_CHUNK_TYPE, data);
-
-        byte[] expected = intToBytes(data.length);
-        assertThat(chunk[0], is(expected[0]));
-        assertThat(chunk[1], is(expected[1]));
-        assertThat(chunk[2], is(expected[2]));
-        assertThat(chunk[3], is(expected[3]));
-    }
-
-    @Test
-    public void testAsChunkCrcIsCalculatedFromTypeAndData() {
-        byte[] data = {5, 7};
-        byte[] chunk = PngEncoderLogic.asChunk(VALID_CHUNK_TYPE, data);
-
-        ByteBuffer byteBuffer = ByteBuffer.allocate(6);
-        byteBuffer.mark();
-        byteBuffer.put(VALID_CHUNK_TYPE.getBytes());
-        byteBuffer.put(data);
-        byteBuffer.reset();
-        int expectedCrc = PngEncoderLogic.getCrc32(byteBuffer);
-
-        byte[] expected = intToBytes(expectedCrc);
-        assertThat(chunk[10], is(expected[0]));
-        assertThat(chunk[11], is(expected[1]));
-        assertThat(chunk[12], is(expected[2]));
-        assertThat(chunk[13], is(expected[3]));
+    public void testCompressionLevelRange() {
+        // Compression level value must be between -1 and 9 inclusive.
+        PngEncoder encoder = new PngEncoder();
+        for (int i=-1; i<10; i++) {
+            encoder.withCompressionLevel(i);
+        }
+        assertThrows(IllegalArgumentException.class, () -> encoder.withCompressionLevel(-2));
+        assertThrows(IllegalArgumentException.class, () -> encoder.withCompressionLevel(10));
```

Comment:
This could be done using parameterized tests, e.g. like this:
```
@ParameterizedTest
@MethodSource("validCompressionLevels")
public void validCompressionLevel(int compressionLevel) {
    assertDoesNotThrow(() -> new PngEncoder().withCompressionLevel(compressionLevel));
}

private static Stream<Arguments> validCompressionLevels() {
    // Compression level value must be between -1 and 9 inclusive.
    return IntStream.rangeClosed(-1, 9).mapToObj(Arguments::arguments);
}

@ParameterizedTest()
@ValueSource(ints = { -2, 10 })
public void invalidCompressionLevel(int compressionLevel) {
    assertThrows(IllegalArgumentException.class, () -> new PngEncoder().withCompressionLevel(compressionLevel));
}
```

I'm not sure whether that's more readable.

It does require adding this dependency to the `pom.xml`:
```
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter-params</artifactId>
    <version>5.7.0</version>
    <scope>test</scope>
</dependency>
```

Reply:
Nice. Let's do parameterized. :)
I found in the JUnit docs that the provider function can return IntStream. That makes it more readable.

Reply:
Nice, that's even better.